### PR TITLE
add patient ssid and site checks

### PR DESCRIPTION
**Pull request by beaugogh**

### CODE REVIEW

**Comment by PiotrZakrzewski** [comment]
Could you add automated tests for the checks? The validation is pretty complex it is easy to break it unintentionally, I did my best to achieve 100% test coverage for ClinicalData validation. It would be good if you could write tests for each validation check for patients as well.

**Comment by beaugogh** [comment]
ok, I will work on the tests:)

File: src/test/java/nl/thehyve/ocdu/validators/PatientDataOcChecksTests.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    }
+
+    @Test
+    public void testSiteExistSuccess() {
+        Subject subject = new Subject();
+        subject.setSsid("1234");
+
+        //sites do not exist
+        subject.setStudy("S_STUDY1");
+        subject.setSite("Sjogren - Sjogren");
```
Comment (PiotrZakrzewski) [inline src/test/java/nl/thehyve/ocdu/validators/PatientDataOcChecksTests.java:176]: I think that OpenClinica encodes site name after "-" in Study identifier of a site. @jacobrousseau Can you confirm? 
That would mean we have to adapt ODM parser to split Study Identifier of Site on "-" and use the later part of it.

Comment (PiotrZakrzewski) [inline src/test/java/nl/thehyve/ocdu/validators/PatientDataOcChecksTests.java:176]: okay Jacob confirmed that we need indeed this Identifier as it is - it is a technical ID of the site we will need for making a SOAP call.

**Pull request merged**